sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -262,19 +270,55 @@ object SpecifiedWindowFrame {
    */
   def defaultWindowFrame(
       hasOrderSpecification: Boolean,
-      acceptWindowFrame: Boolean): SpecifiedWindowFrame = {
+      acceptWindowFrame: Boolean,
+      excludeType: ExcludeType = ExcludeNoOthers): SpecifiedWindowFrame = {
```

Comment:
The default window frame should be `ExcludeNoOthers`. I think we don't need to add this parameter. We can directly add `ExcludeNoOthers` when constructing `SpecifiedWindowFrame` below.